src/main/java/Storage.java

```diff
@@ -0,0 +1,184 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Scanner;
+import java.io.PrintWriter;
+
+/**
+ * Encapsulates the Storage object that is responsible of writing existing Tasks to the local storage.
+ */
+public class Storage {
+    String filePath;
+
+    /**
+     * Constructs the Storage object.
+     * @param filePath Provides the filepath to the local text file as the local storage.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() {
+        ArrayList<Task> taskStorage = new ArrayList<>();
+        try {
+            File f = new File(filePath);
+            Scanner s = new Scanner(f);
+            while (s.hasNextLine()) {
+                String storedLine = s.nextLine();
+                String[] analyseHolder = storedLine.split(", ", 4);
+                String type = analyseHolder[0];
+                String isDone = analyseHolder[1];
+                String description = analyseHolder[2];
+                Date due = new Date();
+                Date start = new Date();
+                Date end = new Date();
+                if (analyseHolder.length == 4) {
+                    String[] timeSplit = analyseHolder[3].split("-");
+                    if (timeSplit.length == 2) {
+                        SimpleDateFormat startFormat = new SimpleDateFormat("dd/MM/yyyy HHmm");
+                        SimpleDateFormat endFormat = new SimpleDateFormat("HHmm");
+                        start = startFormat.parse(timeSplit[0].trim());
+                        end = endFormat.parse(timeSplit[1].trim());
+                    } else {
+                        due = convertStringToDate(analyseHolder[3]);
+                    }
+                }
+
+                Task task = new Task("dummy");
+                Task.reduceTaskCount();
+                if (type.equals("T")) {
+                    task = new Todo(description);
+                } else if (type.equals("D")) {
+                    task = new Deadline(description, due);
+                } else if (type.equals("E")) {
+                    task = new Event(description, start, end);
+                }
+
+                if (isDone.equals("1")) {
+                    task.markAsDone();
+                }
+
+                taskStorage.add(task);
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("The local file cannot be located. " +
```

Comment:
Maybe if local file is not found, create the local file? What if the user is running Duke for the first time? Just a suggestion

Reply:
Thanks for the suggestion!